### ISSUE TITLE
Can't convert UTC tz-aware array to pandas using pd.ArrowDtype

### ISSUE THREAD

**Issue by 0x26res** [apache/arrow#37009]
### Describe the bug, including details regarding any error messages, version, and platform.

I'm trying to convert arrow data of type `pa.timestamp("ns", "UTC")` to pandas, using `pd.ArrowDType`

It works for columns within a table:

```python
df = pa.table(
    {"col1": pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC"))}
).to_pandas(types_mapper=pd.ArrowDtype)
assert df["col1"].dtype == pd.ArrowDtype(pa.timestamp("ns", "UTC"))
```

But it doesn't work for single arrays:

```python
pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC")).to_pandas(
        types_mapper=pd.ArrowDtype
)
```

```
Traceback (most recent call last):
    pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC")).to_pandas(
  File "pyarrow/array.pxi", line 837, in pyarrow.lib._PandasConvertible.to_pandas
  File "pyarrow/array.pxi", line 1446, in pyarrow.lib.Array._to_pandas
  File "pyarrow/array.pxi", line 1679, in pyarrow.lib._array_like_to_pandas
  File "./venv/lib/python3.10/site-packages/pyarrow/pandas_compat.py", line 1264, in make_tz_aware
    series = (series.dt.tz_localize('utc')
  File "./venv/lib/python3.10/site-packages/pandas/core/accessor.py", line 112, in f
    return self._delegate_method(name, *args, **kwargs)
  File "./venv/lib/python3.10/site-packages/pandas/core/indexes/accessors.py", line 199, in _delegate_method
    result = getattr(self._parent.array, f"_dt_{name}")(*args, **kwargs)
  File "./venv/lib/python3.10/site-packages/pandas/core/arrays/arrow/array.py", line 2203, in _dt_tz_localize
    result = pc.assume_timezone(
  File "./venv/lib/python3.10/site-packages/pyarrow/compute.py", line 259, in wrapper
    return func.call(args, options, memory_pool)
  File "pyarrow/_compute.pyx", line 367, in pyarrow._compute.Function.call
  File "pyarrow/error.pxi", line 144, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 100, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Timestamps already have a timezone: 'UTC'. Cannot localize to 'utc'.
```

I found a work around for now:

```
series = (
    pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC"))
    .cast(pa.timestamp("ns"))
    .to_pandas(types_mapper=pd.ArrowDtype)
    .dt.tz_localize("UTC")
)
assert series.dtype == pd.ArrowDtype(pa.timestamp("ns", "UTC"))
```

Full example:

```python
import pyarrow as pa
import pandas as pd
import pytest

df = pa.table(
    {"col1": pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC"))}
).to_pandas(types_mapper=pd.ArrowDtype)
assert df["col1"].dtype == pd.ArrowDtype(pa.timestamp("ns", "UTC"))

pa.array([pd.Timestamp.now()], pa.timestamp("ns")).to_pandas(
    types_mapper=pd.ArrowDtype
)

with pytest.raises(
        pa.ArrowInvalid,
        match=r"Timestamps already have a timezone: 'UTC'. Cannot localize to 'utc'.",
):
    pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC")).to_pandas(
        types_mapper=pd.ArrowDtype
    )

series = (
    pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC"))
    .cast(pa.timestamp("ns"))
    .to_pandas(types_mapper=pd.ArrowDtype)
    .dt.tz_localize("UTC")
)
assert series.dtype == pd.ArrowDtype(pa.timestamp("ns", "UTC"))
```
This could be related to other issues:
- https://github.com/apache/arrow/issues/35633
- https://github.com/aws/aws-sdk-pandas/issues/2410

### Component(s)

Python

**Comment by AlenkaF** [comment]
The errors do seem related to the ones you listed. Could we first check the package versions?
When running on pyarrow dev and pandas 2.0.0 the code seems to work for me:

```python
>>> import pyarrow as pa
>>> import pandas as pd

>>> pa.__version__
'13.0.0.dev670+g757e0d5cc'
>>> pd.__version__
'2.0.0'

>>> df = pa.table(
...     {"col1": pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC"))}
... ).to_pandas(types_mapper=pd.ArrowDtype)
assert df["col1"].dtype == pd.ArrowDtype(pa.timestamp("ns", "UTC"))>>> assert df["col1"].dtype == pd.ArrowDtype(pa.timestamp("ns", "UTC"))
>>> df
                               col1
0  2023-08-17 05:36:07.142786+00:00
>>> df["col1"]
0    2023-08-17 05:36:07.142786+00:00
Name: col1, dtype: timestamp[ns, tz=UTC][pyarrow]

>>> pa.array([pd.Timestamp.utcnow()], pa.timestamp("ns", "UTC")).to_pandas(
...         types_mapper=pd.ArrowDtype
... )
0    2023-08-17 05:36:15.453414+00:00
dtype: timestamp[ns, tz=UTC][pyarrow]
```

Which versions are you using @0x26res?

**Comment by 0x26res** [comment]
I'm using pyarrow==12.0.1 and pandas==2.0.3.

So should we expect this to be fixed in pyarrow 13.0.0?

**Comment by AlenkaF** [comment]
> So should we expect this to be fixed in pyarrow 13.0.0?

Yes. Looking at the issues included into the new release, this next PR seems to fix it:
https://github.com/apache/arrow/issues/36096

**Comment by 0x26res** [comment]
OK, I'll close the issue. I'll try to double check once 13.0.0 is out. Thanks.
